swing-lib/src/main/java/org/triplea/swing/DialogBuilder.java

```diff
@@ -15,8 +16,18 @@
  * yes/no confirmation dialog will be shown to the user, if yes is clicked the confirm action will be executed.
  */
 public final class DialogBuilder {
+  private static boolean uiEnabled = true;
+
   private DialogBuilder() {}
 
+  /**
+   * Method to turn off UI notifications for cases when we are running tests.
+   */
+  @VisibleForTesting
+  public static void disableUi() {
+    uiEnabled = false;
+  }
```

Comment:
This is kinda ugly, but arguably swing is pretty annoying when it comes to writing tests

Reply:
Agree, this is a place where the integ test should be injecting a mock or a fake implementation. Certainly a bit quick and dirty.